File: tests/MethodAttr.pas
namespace Demo;

type
  AttrSample = public class
  public
    class method Foo: Integer; static;
    [Obsolete]
    class method Bar(x: Integer): Integer; static;
  end;

  WebService = public class
  public
    [WebMethod(EnableSession := true)]
    [ScriptMethod]
    class function SalvarCadastro(natrend, prestserv, data, base_calculo, vl_imposto: String): String;
function DescricaoNatRend(natrend: Object): String;
  end;

implementation

class method AttrSample.Foo: Integer;
begin
  Result := 0;
end;

class method AttrSample.Bar(x: Integer): Integer;
begin
  Result := x;
end;

class function WebService.SalvarCadastro(natrend, prestserv, data, base_calculo, vl_imposto: String): String;
begin
  Result := '';
end;

[WebMethod]
function WebService.DescricaoNatRend(natrend: Object): String;
begin
  Result := '';
end;

end.

File: tests/MethodAttr.cs
namespace Demo {
    public partial class AttrSample {
        public static int Foo() {
            int result;
            result = 0;
            return result;
        }
        [Obsolete]
        public static int Bar(int x) {
            int result;
            result = x;
            return result;
        }
    }
    
    public partial class WebService {
        [WebMethod(EnableSession = true)]
        [ScriptMethod]
        public static string SalvarCadastro(string natrend, string prestserv, string data, string base_calculo, string vl_imposto) {
            string result;
            result = "";
            return result;
        }
        [WebMethod]
        public string DescricaoNatRend(object natrend) {
            string result;
            result = "";
            return result;
        }
    }
}
